projects/common/a10gx/a10gx_system_qsys.tcl

```diff
@@ -211,37 +214,55 @@ set_connection_parameter_value sys_cpu.instruction_master/sys_flash.uas arbitrat
-add_connection sys_clk.clk mm_ccb_sys_cpu.s0_clk
-add_connection sys_clk.clk_reset mm_ccb_sys_cpu.s0_reset
-add_connection sys_cpu.data_master mm_ccb_sys_cpu.s0
-add_connection sys_cpu.instruction_master mm_ccb_sys_cpu.s0
-add_connection sys_ddr3_cntrl.emif_usr_clk mm_ccb_sys_cpu.m0_clk
-add_connection sys_ddr3_cntrl.emif_usr_reset_n mm_ccb_sys_cpu.m0_reset
-add_connection  mm_ccb_sys_cpu.m0 sys_ddr3_cntrl.ctrl_amm_0
-
 # cpu/hps handling
 
 proc ad_cpu_interrupt {m_irq m_port} {
 
   add_connection sys_cpu.irq ${m_port}
   set_connection_parameter_value sys_cpu.irq/${m_port} irqNumber ${m_irq}
-}
 
-proc ad_cpu_interconnect {m_base m_port} {
+}
 
-  add_connection sys_cpu.data_master ${m_port}
-  set_connection_parameter_value sys_cpu.data_master/${m_port} baseAddress [expr ($m_base + 0x10000000)]
+proc ad_cpu_interconnect {m_base m_port {avl_bridge ""} {avl_bridge_baseaddr 0x10000000}} {
+
+  if {[string equal ${avl_bridge} ""]} {
+    add_connection sys_cpu.data_master ${m_port}
+    set_connection_parameter_value sys_cpu.data_master/${m_port} baseAddress [expr ($m_base + 0x10000000)]
+  } else {
+    if {[lsearch -exact [get_instances] ${avl_bridge}] == -1} {
+      ## Instantiate the bridge and connect the interfaces
+      add_instance ${avl_bridge} altera_avalon_mm_bridge
+      set_instance_parameter_value ${avl_bridge} {USE_AUTO_ADDRESS_WIDTH} {1}
+      #set_instance_parameter_value ${avl_bridge} {ADDRESS_WIDTH} {17}
+      set_instance_parameter_value ${avl_bridge} {SYNC_RESET} {1}
+      add_connection sys_cpu.data_master ${avl_bridge}.s0
+      set_connection_parameter_value sys_cpu.data_master/${avl_bridge}.s0 baseAddress ${avl_bridge_baseaddr}
+      add_connection sys_clk.clk ${avl_bridge}.clk
+      add_connection sys_clk.clk_reset ${avl_bridge}.reset
+    }
+    add_connection ${avl_bridge}.m0 ${m_port}
+    set_connection_parameter_value ${avl_bridge}.m0/${m_port} baseAddress ${m_base}
+  }
 }
 
 proc ad_dma_interconnect {m_port} {
 
-  add_connection ${m_port} sys_ddr3_cntrl.ctrl_amm_0
-  set_connection_parameter_value ${m_port}/sys_ddr3_cntrl.ctrl_amm_0  baseAddress {0x0}
+  set avm_bridge ""
+  append avm_bridge [lindex [split $m_port "."] 0] "_bridge"
+  set if_name [lindex [split $m_port "."] 1]
+
+  ## Instantiate an Avalon Pipeline Bridge, in order to isolate the AXI to Avalon
+  ## adapters from the main interconnect
+  add_instance ${avm_bridge} altera_avalon_mm_bridge
+  set_instance_parameter_value ${avm_bridge} {SYNC_RESET} {1}
+  set_instance_parameter_value ${avm_bridge} {DATA_WIDTH} {128}
+  set_instance_parameter_value ${avm_bridge} {USE_AUTO_ADDRESS_WIDTH} {1}
+  add_connection sys_clk.clk ${avm_bridge}.clk
+  add_connection sys_clk.clk_reset ${avm_bridge}.reset
+  add_connection ${m_port} ${avm_bridge}.s0
+  add_connection ${avm_bridge}.m0 sys_ddr3_cntrl.ctrl_amm_0
+  set_connection_parameter_value ${avm_bridge}.m0/sys_ddr3_cntrl.ctrl_amm_0 baseAddress {0x0}
+
```

Comment:
Does this mean we need to update the devicetree ? Or the addresses remain the same ?

Reply:
No need to update the device tree. But because we don't actually use this at the moment, will not add it into this PR. If in the future we will have timing issues with the main interconnect, we can revisit this solution.